### PR TITLE
Swath

### PR DESCRIPTION
**Pull request by vmusch**
# Description

- OpenMS Matrix data structure added to OpenSwath
- MRMScoring of Openswath revised
- MRMScoring test extended

Please include a summary of the change and which issue is fixed.

# Checklist:
- [ x] Make sure that you are listed in the AUTHORS file
- [ x] Add relevant changes and new features to the CHANGELOG file
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] New and existing unit tests pass locally with my changes
- [ x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by hroest** [comment]
is Matrix.h just duplicated from OpenMS?

**Comment by cbielow** [comment]
Yes, though without the openms dependencies.

**Comment by vmusch** [comment]
rebuild jenkins

File: src/pyOpenMS/pxds/MRMScoring.pxd [hunk 1]
```diff
@@ -7,10 +7,11 @@ from ISpectrumAccess cimport *
 # from StatsHelpers cimport *
 # from Scoring cimport *
 from LightTargetedExperiment cimport *
+from Matrix cimport *
 
-cdef extern from "<OpenMS/OPENSWATHALGO/ALGO/MRMScoring.h>" namespace "OpenSwath":
+cdef extern from "<OpenMS/ANALYSIS/OPENSWATH/MRMScoring.h>" namespace "OpenSwath":
     
-    cdef cppclass MRMScoring "OpenSwath::MRMScoring":
+    cdef cppclass MRMScoring "OpenMS::MRMScoring":
```
Comment (cbielow) [inline src/pyOpenMS/pxds/MRMScoring.pxd:14]: should be `cdef cppclass MRMScoring MRMScoring:`  ???

**Comment by cbielow** [comment]
tests pass.
Ready for merge.

Duplication of Matrix class was undone. Makes it a lot cleaner.

**Comment by cbielow** [comment]
rebuild jenkins

**Review by timosachsenberg — CHANGES_REQUESTED** [review]
Thanks! only minor comments.
Maybe extend the PR description a bit more why the changes have been made and which changes have been made.
This helps immensely if we need to go back to check why some changes have been made.

File: src/tests/class_tests/openswathalgo/MRMScoring_test.cpp [hunk 2]
```diff
@@ -121,7 +121,7 @@ void fill_mock_objects2(MockMRMFeature * imrmfeature, std::vector<std::string>&
 
   std::vector<double> ms1intensity1 {0.0, 110.0, 200.0, 270.0, 320.0, 350.0, 360.0, 350.0, 320.0, 270.0, 200.0};
 
-  std::vector<double> ms1intensity2 {0.0, 115.0, 180.0, 280.0, 330.0, 340.0, 390.0, 320.0, 300.0, 250.0, 100.0};
+  std::vector<double> ms1intensity2 {10.0, 115.0, 180.0, 280.0, 330.0, 340.0, 390.0, 320.0, 300.0, 250.0, 100.0};
```
[the remark this answers is not in the document]
Comment (vmusch) [reply src/tests/class_tests/openswathalgo/MRMScoring_test.cpp]: https://github.com/OpenMS/OpenMS/blob/7dc444e43a81508e19b5c4c8d9eb726aec1f9ae5/src/tests/class_tests/openswathalgo/MRMScoring_test.cpp#L135
I had made a mistake when copying the line (forgot a one)

**Comment by hroest** [comment]
> Duplication of Matrix class was undone. Makes it a lot cleaner.

however now part of the algorithm part got moved out of OpenSwathAlgo, raising the question how much purpose this really still serves

**Comment by timosachsenberg** [comment]
> Maybe extend the PR description a bit more why the changes have been made and which changes have been made.
> This helps immensely if we need to go back to check why some changes have been made.

maybe @vmusch or @cbielow can add a few lines why this has been done

**Comment by timosachsenberg** [comment]
@vmusch @cbielow could your changes here be related to 
https://github.com/OpenMS/OpenMS/issues/5668

**Comment by cbielow** [comment]
not sure. The easiest test would be to try some older commit and see if the problem persists.
Even OpenMS 2.7 should do?!

**Pull request merged**